src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -428,7 +435,17 @@ int IDebugPendingBreakpoint2.Delete()
                 _deleted = true;
                 if (_engine.DebuggedProcess.ProcessState != ProcessState.Stopped && !_engine.DebuggedProcess.MICommandFactory.AllowCommandsWhileRunning())
                 {
-                    _pendingDelete = true;
+                    if (!PendingDeletesEnabled() && _bp != null)
```

Comment:
>PendingDeletesEnabled [](start = 25, length = 21)

I am not sure I understand why this change is being made. Is this because we want to make sure we delete the breakpoint before we set another? If so, would it instead make sense to call DeleteBreakpointsPendingDeletion before setting one when `RequireHardwareBreakpoints` is set? Or are there other reasons why it is important to block waiting on the delete to happen?

Reply:
Yes in answer to your first question: we need to make sure it's deleted before setting another so that GDB knows how many hardware breakpoint registers are actually occupied (as opposed to just being squatted on by a breakpoint pending delete). There is no other reason to block waiting for the delete. I reverted the logic here and instead placed a call to `_bpManager.DeleteBreakpointsPendingDeletion()` at the top of `BindAsync()`. Is this what you had in mind?

Reply:
Yup